src/qibo/models/error_mitigation.py

```diff
@@ -1168,17 +1164,23 @@ def _execute_circuit(circuit, qubit_map, noise_model=None, nshots=10000, backend
     Returns:
         qibo.states.CircuitResult: The result of the circuit execution.
     """
+    from qibo.transpiler.pipeline import Passes
     from qibo.transpiler.placer import Custom
 
-    # TODO: remove backend.platform.topology and pragma: no cover
     if backend is None:  # pragma: no cover
         backend = get_backend()
-    elif get_backend_name() == "qibolab":  # pragma: no cover
-        transpiler = Custom(
-            initial_map=qubit_map, connectivity=backend.platform.topology
+    elif backend.name == "qibolab":  # pragma: no cover
+        qubits = backend.qubits
+        connectivity_edges = backend.connectivity
+        node_mapping = {q: i for i, q in enumerate(qubits)}
+        edges = [(node_mapping[e[0]], node_mapping[e[1]]) for e in connectivity_edges]
+        connectivity = nx.Graph(edges)
```

Comment:
This is now repeated, here and in the global.

I'd propose you, even in this PR, to split this as a standalone function in the transpiler (converting a list of pairs into a `networkx.Graph` object), and applying it directly in the transpiler classes, such that outside you can keep using lists of pairs, without having to take into account the need for `networkx` conversion.

Reply:
This code will be removed once we update the transpiler. They are only required for the current `i:q{i}` mapping. Should we update it now?

Reply:
> This code will be removed once we update the transpiler. They are only required for the current `i:q{i}` mapping. Should we update it now?

If now, I would just do this one and nothing else. Otherwise, start writing the issue for the transpiler refactor, and embed this as an item.

Reply:
I will also leave it as is for now.